### ISSUE TITLE
Implement non-linear modelling using halo bias

### ISSUE THREAD

**Issue by slosar** [LSSTDESC/firecrown#80]
This is a sister ticket to https://github.com/LSSTDESC/firecrown/issues/52 .
Implement halo model fitting into firecrown. Not that I think this is what we will ultimately use, but it would be useful for tests of model sufficiency, etc. Useful for redoing HSC fits etc.
David can probably do this while walking his dog.

**Comment by damonge** [comment]
A tunable halo model has just been merged into CCL.
I'd need to think a bit about how much of this (e.g. HODs) should go into firecrown and how much into CCL. I'm leaning towards CCL or sister modules.

**Comment by slosar** [comment]
Yes, all predictions should go into CCL. Firecrown will just do the plumbing.

**Comment by slosar** [comment]
See also https://github.com/LSSTDESC/firecrown/issues/47 . In my mind, assembly bias goes as an extension of halo model, though not sure we need in the first instance.
